medusa/server/web/core/base.py

```diff
@@ -226,10 +226,13 @@ def redirect(self, url, permanent=False, status=None):
                                             utf8(url)))
 
     def get_current_user(self):
-        user_cookie = self.get_secure_cookie(app.SECURE_TOKEN)
-        if user_cookie:
-            return json.dumps(user_cookie)
-        return None
+        if not isinstance(self, UI):
+            if app.WEB_USERNAME and app.WEB_PASSWORD:
+                return self.get_secure_cookie('medusa_user')
+            else:
+                return True
+        else:
+            return None
```

Comment:
@OmgImAlexis with this (reverted old way + add user/pass check as second if, when you access "/ui/add_message/?a" without being logged im redirected to loging page and not the page you get here:  https://pymedusa.slack.com/files/xo/F2PSXMU9E/screen_shot_2016-10-16_at_12.22.37_pm.png